hamocc/mo_apply_rivin.F90

```diff
@@ -0,0 +1,162 @@
+!  *REAL*      *omask*   - ocean mask
+!  *REAL*      *rivin*   - riverine input field [kmol m-2 yr-1]
+!
+!--------------------------------------------------------------------------------
+  use mo_control_bgc, only: dtb,do_rivinpt
+  use mo_param1_bgc,  only: nriv,irdin,irdip,irsi,iralk,iriron,irdoc,irdet,     &
+                            iano3,iphosph,isilica,isco212,iiron,idoc,idet,      &
+                            ialkali,inatsco212,inatalkali
+  use mo_vgrid,       only: kmle
+  use mo_carbch,      only: ocetra,rivinflx
+
+  implicit none
+
+  integer,intent(in) :: kpie,kpje,kpke
+  real,   intent(in) :: pddpo(kpie,kpje,kpke)
+  real,   intent(in) :: omask(kpie,kpje)
+  real,   intent(in) :: rivin(kpie,kpje,nriv)
+
+  ! local variables
+  integer            :: i,j,k
+  real               :: fdt,volij
+
+
+  ! rivinflx stores the applied n-deposition flux for inventory calculations 
+  ! and output
+  rivinflx(:,:,:) = 0.0
+
+  if (.not. do_rivinpt) return
+  
+!$OMP PARALLEL DO PRIVATE(i,k,fdt,volij)
+  DO j=1,kpje
+  DO i=1,kpie
+    IF(omask(i,j).GT.0.5) THEN
+
+      fdt   = dtb/365.
+
+      ! Distribute riverine inputs over the model mixed layer
+      volij = 0.
+      DO k=1,kmle
+        volij=volij+pddpo(i,j,k)
+      ENDDO
+
+      ! Notes:
+      ! 1) DIC is updated using the assumtions that a_t=a_c+a_n and DIC=a_c (a_t: total 
+      ! alkalinity, a_c: carbonate alkalinity, a_n: contribution of nutrients to a_t). 
+      ! 2) Approx. 80-99% of dFe input is lost to the particulate phase in estuaries at 
+      ! low salinities [Boyle et al., 1977; Chester, 1990; Dai and Martin, 1995; Lohan 
+      ! and Bruland, 2006; Sholkovitz, 1978]. Below we assume 99% loss.
+      ocetra(i,j,1:kmle,iano3)      = ocetra(i,j,1:kmle,iano3)      + rivin(i,j,irdin)*fdt/volij
+      ocetra(i,j,1:kmle,iphosph)    = ocetra(i,j,1:kmle,iphosph)    + rivin(i,j,irdip)*fdt/volij
+      ocetra(i,j,1:kmle,isilica)    = ocetra(i,j,1:kmle,isilica)    + rivin(i,j,irsi) *fdt/volij
+      ocetra(i,j,1:kmle,isco212)    = ocetra(i,j,1:kmle,isco212)    + rivin(i,j,iralk)*fdt/volij  &
+                                                                    + rivin(i,j,irdin)*fdt/volij  &
+                                                                    + rivin(i,j,irdip)*fdt/volij
+      ocetra(i,j,1:kmle,ialkali)    = ocetra(i,j,1:kmle,ialkali)    + rivin(i,j,iralk)*fdt/volij
+#ifdef natDIC
+      ocetra(i,j,1:kmle,inatsco212) = ocetra(i,j,1:kmle,inatsco212) + rivin(i,j,iralk)*fdt/volij  &
+                                                                    + rivin(i,j,irdin)*fdt/volij  &
+                                                                    + rivin(i,j,irdip)*fdt/volij
+      ocetra(i,j,1:kmle,inatalkali) = ocetra(i,j,1:kmle,inatalkali) + rivin(i,j,iralk)*fdt/volij
+#endif
+      ocetra(i,j,1:kmle,iiron)      = ocetra(i,j,1:kmle,iiron)      + rivin(i,j,iriron)*fdt/volij*0.01
```

Comment:
With the fully new restructuring, I feel it would be nice to replace the explicit value of 0.01 by a parameter value, but we can do this later as well.